### ISSUE TITLE
test failures on ubuntu systems using dash

### ISSUE THREAD

**Issue by tmm1** [rtomayko/posix-spawn#6]
vagrant@vagrantup:~/code/posix-spawn$ ls -alh /bin/sh
    lrwxrwxrwx 1 root root 4 2010-07-20 23:02 /bin/sh -> dash

    vagrant@vagrantup:~/code/posix-spawn$ rake
    (in /home/vagrant/code/posix-spawn)
    cp tmp/x86_64-linux/posix_spawn_ext/1.8.7/posix_spawn_ext.so lib/posix_spawn_ext.so
    Loaded suite /usr/lib/ruby/gems/1.8/gems/rake-0.8.7/lib/rake/rake_test_loader
    Started
    ...F..................exec: 1: 101: not found
    FF..FFFFF...........................exec: 1: 101: not found
    FF..FFFFF...................................
    Finished in 1.347913 seconds.

      1) Failure:
    test_backtick_redirect(BacktickTest) [./test/test_backtick.rb:27]:
    <"/bin/sh: nosuchcmd: not found\n"> expected but was
    <"/bin/sh: nosuchcmd: command not found\n">.

      2) Failure:
    test_explicitly_passing_an_fd_as_open(ForkSpawnTest)
        [./test/test_spawn.rb:332:in `assert_process_exit_status'
         ./test/test_spawn.rb:325:in `assert_process_exit_ok'
         ./test/test_spawn.rb:185:in `test_explicitly_passing_an_fd_as_open']:
    <0> expected but was
    <2>.

      3) Failure:
    test_sanity_of_checking_clone_with_sh(ForkSpawnTest)
        [./test/test_spawn.rb:332:in `assert_process_exit_status'
         ./test/test_spawn.rb:72:in `test_sanity_of_checking_clone_with_sh']:
    <0> expected but was
    <2>.

      4) Failure:
    test_spawn_close_on_standard_stream_io_object(ForkSpawnTest)
        [./test/test_spawn.rb:332:in `assert_process_exit_status'
         ./test/test_spawn.rb:88:in `test_spawn_close_on_standard_stream_io_object']:
    <1> expected but was
    <2>.

      5) Failure:
    test_spawn_close_option_with_fd_number(ForkSpawnTest)
        [./test/test_spawn.rb:332:in `assert_process_exit_status'
         ./test/test_spawn.rb:98:in `test_spawn_close_option_with_fd_number']:
    <1> expected but was
    <2>.

      6) Failure:
    test_spawn_close_option_with_io_object(ForkSpawnTest)
        [./test/test_spawn.rb:332:in `assert_process_exit_status'
         ./test/test_spawn.rb:109:in `test_spawn_close_option_with_io_object']:
    <1> expected but was
    <2>.

      7) Failure:
    test_spawn_close_option_with_symbolic_standard_stream_names(ForkSpawnTest)
        [./test/test_spawn.rb:332:in `assert_process_exit_status'
         ./test/test_spawn.rb:79:in `test_spawn_close_option_with_symbolic_standard_stream_names']:
    <1> expected but was
    <2>.

      8) Failure:
    test_spawn_closing_multiple_fds_with_array_keys(ForkSpawnTest)
        [./test/test_spawn.rb:332:in `assert_process_exit_status'
         ./test/test_spawn.rb:127:in `test_spawn_closing_multiple_fds_with_array_keys']:
    <1> expected but was
    <2>.

      9) Failure:
    test_explicitly_passing_an_fd_as_open(PosixSpawnTest)
        [./test/test_spawn.rb:332:in `assert_process_exit_status'
         ./test/test_spawn.rb:325:in `assert_process_exit_ok'
         ./test/test_spawn.rb:185:in `test_explicitly_passing_an_fd_as_open']:
    <0> expected but was
    <2>.

     10) Failure:
    test_sanity_of_checking_clone_with_sh(PosixSpawnTest)
        [./test/test_spawn.rb:332:in `assert_process_exit_status'
         ./test/test_spawn.rb:72:in `test_sanity_of_checking_clone_with_sh']:
    <0> expected but was
    <2>.

     11) Failure:
    test_spawn_close_on_standard_stream_io_object(PosixSpawnTest)
        [./test/test_spawn.rb:332:in `assert_process_exit_status'
         ./test/test_spawn.rb:88:in `test_spawn_close_on_standard_stream_io_object']:
    <1> expected but was
    <2>.

     12) Failure:
    test_spawn_close_option_with_fd_number(PosixSpawnTest)
        [./test/test_spawn.rb:332:in `assert_process_exit_status'
         ./test/test_spawn.rb:98:in `test_spawn_close_option_with_fd_number']:
    <1> expected but was
    <2>.

     13) Failure:
    test_spawn_close_option_with_io_object(PosixSpawnTest)
        [./test/test_spawn.rb:332:in `assert_process_exit_status'
         ./test/test_spawn.rb:109:in `test_spawn_close_option_with_io_object']:
    <1> expected but was
    <2>.

     14) Failure:
    test_spawn_close_option_with_symbolic_standard_stream_names(PosixSpawnTest)
        [./test/test_spawn.rb:332:in `assert_process_exit_status'
         ./test/test_spawn.rb:79:in `test_spawn_close_option_with_symbolic_standard_stream_names']:
    <1> expected but was
    <2>.

     15) Failure:
    test_spawn_closing_multiple_fds_with_array_keys(PosixSpawnTest)
        [./test/test_spawn.rb:332:in `assert_process_exit_status'
         ./test/test_spawn.rb:127:in `test_spawn_closing_multiple_fds_with_array_keys']:
    <1> expected but was
    <2>.

    102 tests, 298 assertions, 15 failures, 0 errors
    rake aborted!
    Command failed with status (1): [/usr/bin/ruby1.8 -I"lib" "/usr/lib/ruby/ge...]

    (See full trace by running task with --trace)

**Comment by rtomayko** [comment]
This was fixed:

```
[rtomayko@enterprise2-dev7-cp1-prd:posix-spawn]$ ls -l /bin/sh
lrwxrwxrwx 1 root 4 Mar 29  2012 /bin/sh -> dash
[rtomayko@enterprise2-dev7-cp1-prd:posix-spawn]$ rake
cp tmp/x86_64-linux/posix_spawn_ext/2.1.5/posix_spawn_ext.so lib/posix_spawn_ext.so
Run options: --seed 2197

# Running:

................................................................................................................................................

Finished in 0.687988s, 209.3060 runs/s, 672.9768 assertions/s.

144 runs, 463 assertions, 0 failures, 0 errors, 0 skips
```

**Comment by piki** [comment]
Fixed in #58
